tests/publish/aspects.js

```diff
@@ -0,0 +1,685 @@
+              const evtBody = evt[aspectEvents.add];
+              expect(evtBody).to.include.keys('createdAt', 'createdBy',
+                'criticalRange', 'description', 'helpEmail', 'helpUrl', 'id',
+                'imageUrl', 'infoRange', 'isPublished', 'name', 'okRange',
+                'ownerId', 'rank', 'relatedLinks', 'tags', 'timeout',
+                'updatedAt', 'valueLabel', 'valueType', 'warningRange');
+              expect(evtBody).to.have.property('name', name);
+              expect(evtBody).to.have.property('isPublished', true);
+              done();
+            });
+        });
+    });
+
+    it('update name of published aspect, subscriber gets remove event ' +
+      'then add event', (done) => {
+      const name = `${tu.namePrefix}PatchName`;
+      const newName = `${tu.namePrefix}NewName`;
+
+      api.post(path)
+        .set('Authorization', token)
+        .send({ name, timeout: '1m', isPublished: true })
+        .expect(constants.httpStatus.CREATED)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(subscribeTracker).to.have.length(1);
+          const evt = JSON.parse(subscribeTracker[0]);
+          expect(evt).to.have.property(aspectEvents.add);
+          subscribeTracker = [];
+
+          api.patch(`${path}/${name}`)
+            .set('Authorization', token)
+            .send({ name: newName })
+            .expect(constants.httpStatus.OK)
+            .end((_err, _res) => {
+              if (_err) {
+                return done(err);
+              }
+
+              expect(subscribeTracker).to.have.length(2);
+              const evt0 = JSON.parse(subscribeTracker[0]);
+              const evt1 = JSON.parse(subscribeTracker[1]);
+              expect(evt0).to.have.property(aspectEvents.del);
+              const evtBody0 = evt0[aspectEvents.del];
+              expect(evtBody0).to.include.keys('createdAt', 'createdBy',
+                'criticalRange', 'description', 'helpEmail', 'helpUrl', 'id',
+                'imageUrl', 'infoRange', 'isPublished', 'name', 'okRange',
+                'ownerId', 'rank', 'relatedLinks', 'tags', 'timeout',
+                'updatedAt', 'valueLabel', 'valueType', 'warningRange');
+              expect(evtBody0).to.have.property('name', name);
+              const evtBody1 = evt1[aspectEvents.add];
+              expect(evtBody1).to.include.keys('createdAt', 'createdBy',
+                'criticalRange', 'description', 'helpEmail', 'helpUrl', 'id',
+                'imageUrl', 'infoRange', 'isPublished', 'name', 'okRange',
+                'ownerId', 'rank', 'relatedLinks', 'tags', 'timeout',
+                'updatedAt', 'valueLabel', 'valueType', 'warningRange');
+              expect(evtBody1).to.have.property('name', newName);
+              done();
+            });
+        });
+    });
+
+    it('update tags of published aspect, subscriber gets update event',
+      (done) => {
```

Comment:
For subjects, we send a delete and then add when tags are changed. Do we need to do that for aspects as well?

Reply:
Oh, I think we do need to do that for filtering... if we don't get the remove+add then if a tag change results in the aspect no longer being relevant for that perspective, it would never get the signal to remove. I will change that.

Reply:
Also need to update test description.